include/swift/AST/DiagnosticsParse.def

```diff
@@ -1768,6 +1768,13 @@ WARNING(attr_warn_unused_result_removed,none,
         "'warn_unused_result' attribute behavior is now the default", ())
 ERROR(attr_warn_unused_result_expected_rparen,none,
       "expected ')' after 'warn_unused_result' attribute", ())
+      
+// warn
+ERROR(attr_warn_expected_diagnostic_group_identifier,none,
+      "expected '%0' option to be a diagnostic group identifier", (StringRef))
```

Comment:
Do we want this one to be an error or should it just be a warning? IIUC an error would make it source breaking to remove a diagnostic group and could break source backward compatibility when adding this attribute for a new diagnostic group.

In the same sense, if we want to be more clear about the diagnostic group being unknown by the current compiler we could use something like `the diagnostic group identifier '%0' is unknown`.

Reply:
That's a great point, changed to a warning. Thank you.